### ISSUE TITLE
Extension should only run on LaTeX builder

### ISSUE THREAD

**Issue by Daltz333** [missinglinkelectronics/sphinxcontrib-svg2pdfconverter#8]
The extension currently runs on all builders, HTML, Gettext, Linkcheck, etc.

**Comment by sephalon** [comment]
The SVG to PDF converters are based on the Sphinx ``ImageConverter`` class that is only invoked [if the builder does not support the image type](https://github.com/sphinx-doc/sphinx/blob/cfadd5f3f342a189b86f1a873252fe21fd19725d/sphinx/transforms/post_transforms/images.py#L203-L204). So this should not happen – could you please elaborate a bit more on what exactly you see?

**Comment by Daltz333** [comment]
Reopening as this issue is still present in ``sphinx-build -b html``. rsvg-convert should *only* run in the latex builder

**Comment by sephalon** [comment]
https://github.com/sphinx-doc/sphinx/pull/7978 has been merged by now. Does it solve the issue for you?

**Comment by Daltz333** [comment]
No. All that issue does is prevent it from running many times on those builders. This extension still runs on all builders.

**Comment by sephalon** [comment]
sphinx-doc/sphinx#8649 should finally fix this issue. Closed.

**Comment by mgeier** [comment]
I'm still getting an error, is this really solved?

See https://github.com/spatialaudio/nbsphinx/pull/559 for details.

**Comment by Daltz333** [comment]
Works for me. That error you have looks like it's having an issue running rsvg-convert. Try running it by hand and verifying it works.

**Comment by mgeier** [comment]
Thanks for the quick answer @Daltz333!
I was expecting that `rsvg-convert` wouldn't be called at all when not needed, isn't that so?

I'm only generating HTML in the linked issue, so I would expect that `rsvg-convert` is not needed and doesn't have to be called.

I have intentionally *not* installed it for this situation.

**Comment by Daltz333** [comment]
Hm, are you using a custom builder to the standard HTML builder?

**Comment by mgeier** [comment]
Well I'm using some custom nodes and quite a few suspicious hacks in `nbsphinx`, but I think I'm using the standard HTML builder.

**Comment by Daltz333** [comment]
I attempted to build your documentation locally, but looks like it has several things line-endings related that prevents me (Windows) from building it.

This is the logic doing the skip
```python
elif set(node['candidates']) & set(self.app.builder.supported_image_types):
            # builder supports the image; no need to convert
            return False
```

**Comment by mgeier** [comment]
I'm not changing anything regarding `builder.supported_image_types`, at least not knowingly.

> looks like it has several things line-endings related that prevents me (Windows) from building it.

It would be great if you could create an issue for this!
I'd like to make it work on Windows, but I'm not a user myself ...

**Comment by mgeier** [comment]
@Daltz333 I looked into this again and I made a minimal reproducible example: https://github.com/mgeier/rsvgconverter-bug

At first I couldn't reproduce the problem, but then I found out that the error occurs when using a "data" URL instead of an image file, e.g.

```rst
.. image:: data:image/png;base64,....
```

I've added this to my test repo: https://github.com/mgeier/rsvgconverter-bug/commit/82379d10ee8e5f6ed7fb4cdf7dceb38748f5928e.

Here's a failing test: https://github.com/mgeier/rsvgconverter-bug/runs/3274422618

Any idea how to fix this?
